src/transformers/models/fast/image_processing_fast.py

```diff
@@ -0,0 +1,669 @@
+
+        all_images = []
+        for image in images:
+            if do_resize:
+                image = self.resize(image=image, size=size, resample=resample, input_data_format=input_data_format)
+
+            if do_center_crop:
+                image = self.center_crop(image=image, size=crop_size, input_data_format=input_data_format)
+
+            if do_rescale:
+                image = self.rescale(image=image, scale=rescale_factor, input_data_format=input_data_format)
+
+            if do_normalize:
+                image = self.normalize(
+                    image=image, mean=image_mean, std=image_std, input_data_format=input_data_format
+                )
+
+            all_images.append(image)
+        images = [
+            to_channel_dimension_format(image, data_format, input_channel_dim=input_data_format)
+            for image in all_images
+        ]
+
+        data = {"pixel_values": images}
+        return BatchFeature(data=data, tensor_type=return_tensors)
+
+    def _max_pooling(self, input_tensor, scale=1):
+        kernel_size = self.pooling_size // 2 + 1 if scale == 2 else self.pooling_size
+        padding = (self.pooling_size // 2) // 2 if scale == 2 else (self.pooling_size - 1) // 2
+
+        pooling = nn.MaxPool2d(kernel_size=kernel_size, stride=1, padding=padding)
+
+        pooled_output = pooling(input_tensor)
+        return pooled_output
+
+    def post_process_text_detection(self, output, target_sizes, threshold, bounding_box_type="rect", image_size=None):
+        """
+        Post-processes the raw model output to generate bounding boxes and scores for text detection.
+
+        Args:
+            output (dict): Dictionary containing model outputs. Must include key `"last_hidden_state"` (Tensor of shape [B, C, H, W]).
+            target_sizes (List[Tuple[int, int]]): Original image sizes (height, width) for each item in the batch.
+                                                Used to scale detection results back to original image dimensions.
+            threshold (float): Confidence threshold for filtering low-score text regions.
+            bounding_box_type (str, optional): Type of bounding box to return. Must be one of:
+                                    - "rect": rotated bounding rectangles
+                                    - "poly": polygon boundaries
+            image_size (Tuple[int, int], optional): Size (height, width) of the image used during inference.
+                                                If not provided, defaults to `self.img_size`.
+
+        Returns:
+            List[Dict]: A list of dictionaries, each containing:
+                - "boxes" (np.ndarray): Array of detected bounding boxes in the specified format.
+                - "scores" (np.ndarray): Corresponding confidence scores for each bounding box.
+        """
+        scale = 2
+        image_size = image_size if image_size is not None else self.img_size
```

Comment:
Why not to use target_sizes here? I think intermediate image size is not needed here.

Usually we have the following pipeline:

1. Apply postprocessing which is OK for batch
2. Iterate over images -> resize to original size -> apply per-image postprocessing such as resize and following

In addition, target sizes is usually optional, so make sure postprocessing works without `target_sizes` provided as well